TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Command/NodeCommandControllerPlugin.php

```diff
@@ -856,6 +861,95 @@ protected function removeNode($nodeIdentifier, $dimensionsHash)
     }
 
     /**
+     * Remove nodes with invalid dimension values
+     *
+     * This removes nodes which have dimension values not fitting to the current dimension configuration
+     *
+     * @param NodeType $nodeType Only for this node type, if specified
+     * @param string $workspaceName Name of the workspace to consider
+     * @param boolean $dryRun Simulate?
+     * @return void
+     */
+    public function removeNodesWithInvalidDimensions(NodeType $nodeType = null, $workspaceName, $dryRun)
+    {
+        $allowedDimensionCombinations = $this->contentDimensionCombinator->getAllAllowedCombinations();
+
+        if ($nodeType !== null) {
+            $this->output->outputLine('Checking nodes of type "%s" for invalid dimensions ...', array($nodeType->getName()));
+            $nodesArray = $this->collectNodesWithInvalidDimensions($nodeType, $workspaceName, $allowedDimensionCombinations);
+        } else {
+            $this->output->outputLine('Checking for nodes with invalid dimensions ...');
+            $nodesArray = [];
+            foreach ($this->nodeTypeManager->getNodeTypes() as $nodeType) {
```

Comment:
That seems a bit complicated, can we not simply grab all nodes regardless of nodeType in this case?